### DISCUSSION TITLE
Merkle Air Drop Proofs

### DISCUSSION THREAD

**Discussion by Gypsaman** [Cyfrin/foundry-full-course-cu#3084] [Q&A]
if this were a live contract, how does a user obtain the proofs needed to claim the airdrop?

**Comment by EngrPips** [comment]
You will basically have a front end where you can provide those information or maybe you had initially collected the user's mails and emailed them the information.

**Comment by Derastephh** [comment]
sorry but i am not satisfied with the answer here. I have claimed airdrops and all i do is just connect my wallet and claim. However, i dont understand what happens in the background. Please @PatrickAlphaC  in what case do we submit proofs during an airdrop claim. I want to fully understand this before the first flight contest tomorrow.

**Reply by EngrPips** [reply]
Not all Airdrop contracts implement a Merkle Tree. Some use a mapping of address to bool and check that to enable people to claim or not. If they implement a Merkle tree then you will to provide your proof when you want to claim.

**Reply by Derastephh** [reply]
Oh alright. Thanks

**Reply by EngrPips** [reply]
Any time, Friend.
